edge-hub/watchdog/linux/src/main.rs

```diff
@@ -0,0 +1,114 @@
+mod child_process;
```

Comment:
```suggestion
mod child;
```
Coming from a c# it is hard to abandon the habit to make VeryLongAndDetailed types and module names even nit: where it is not necessary. What I noticed working with rust codebases, rules are much more relaxed. 
IMO watchdog is always will monitor processes only, so maybe "_process" part in  `child_process` is not necessary. 
Same for `fn run_child_process()` it the function is inside `child` modules it can be called only `run` without loosing any meanings.